python/cuda_parallel/cuda/parallel/experimental/_utils/protocols.py

```diff
@@ -61,3 +61,31 @@ def is_contiguous(arr: DeviceArrayLike) -> bool:
     else:
         # not contiguous
         return False
+
+
+def _validate_and_get_stream(stream) -> Optional[int]:
+    # null stream is allowed
+    if stream is None:
+        return None
+
+    try:
+        stream_property = stream.__cuda_stream__()
+    except AttributeError as e:
+        raise TypeError(
+            f"stream argument {stream} does not implement the '__cuda_stream__' protocol"
+        ) from e
+
+    try:
+        version = stream_property[0]
+        handle = stream_property[1]
```

Comment:
Any reason to not do

```python
version, handle = stream_property
```

Note that this will raise `ValueError` (not `IndexError` if there are not enough values in `stream_property`.

And if we're concerned about future versions of the protocol containing more items:

```python
version, handle, _* = stream_property
```

Reply:
I don't know we can do partial unpacking this way, very neat!